### PR TITLE
gnuplot: fixed broken package and add variants

### PR DESCRIPTION
**Pull request by davydden**
there is an issue with `ncurses` (or `gnuplot`?) that neeeds to be fixed:
```
In file included from $SPACK_ROOT/darwin-sierra-x86_64/clang-8.0.0-apple/ncurses-6.0-f5ft4ie4brbq44gkyjhr7r42xnfjkhgt/include/term.h:86:
In file included from /usr/include/termios.h:27:
/usr/include/sys/termios.h:263:1: error: expected expression
typedef unsigned long   tcflag_t;
^
```
~~but at least `spack install gnuplot~x11~gd~cairo` works now on macOS.~~
Same happens on Ubuntu16.04.2 with gcc5.4.0.

p.s. looking at how [homebrew builds ncurses](https://github.com/Homebrew/homebrew-dupes/blob/master/ncurses.rb) or [macports](https://github.com/macports/macports-ports/blob/master/devel/ncurses/Portfile), i don't see anything special. Moreover, current args in `ncurses` in Spack are almost identical to those in macports.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gnuplot/package.py [hunk 1]
```diff
@@ -43,24 +41,93 @@ class Gnuplot(Package):
...
+    version('5.0.5', 'c5e96fca73afbee4f57cbc1bfce6b3b8')
     version('5.0.1', '79b4f9e203728f76b60b28bcd402d3c7')
 
+    variant('wx',      default=False,
+            description='Activates wxWidgets terminal')
+    variant('gd',      default=True,
+            description='Activates gd based terminal')
+    variant('cairo',   default=True,
+            description='Activates cairo based terminal')
+    variant('x11',     default=True,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: We usually call this variant `+X`, right? I don't care which we use, but we should probably be consistent across packages. Also, this should default to False like it does in every other package, at least until the concretizer is fixed.

**Comment by davydden** [comment]
@adamjstewart @tgamblin @alalazo it looks like there is a conflict in headers between `ncurses` and `gnuplot` and gnuplot fails to build because compiler picks up the wrong header (thanks to the users of Gnuplot mailing list who pointed this out). This happens even with `transitive_rpaths = False`. I will try to see where include flags are added based on DAG, maybe there is a bug and the flag is disregarded...

**Comment by davydden** [comment]
@tgamblin quick question: Rpaths are added in `build_environment.py` in `get_rpaths`, but where include flags from DAG are being added for compiler wrappers?

**Comment by davydden** [comment]
@alalazo thanks, i will see if I can fix this...

**Comment by tgamblin** [comment]
@davydden: do the macports and homebrew builds transitively include dependencies' include paths?  Spack does.  It may be that we want to stop doing this, although it's not clear to me how homebrew and macports wouldn't run into this problem if they link/include their dependencies' headers in a single prefix... they'd have both `term.h` and `termios.h` too.

Currently the `link` deptype implies adding `-I`, both for immediate and transitive dependencies.  We could potentially solve the issue here by adding `-I`'s ONLY for immediate dependencies.  However, I worry about that because some libraries' headers include the headers of their dependencies, and I think that making the behavior non-transitive (as we allow for `RPATH`s), will break a lot of packages.

If it really is a header conflict, which header do we need to prefer?  If we can figure that out we can talk about how the dependency relationship should be specified.

**Comment by davydden** [comment]
@tgamblin 
I think neither homebrew nor macports do anything with transitive dependencies.

The conflict here is between term.h in ncurses and src/term.h in Gnuplot which is included during documentation build. I don't think this header is installed into prefix/include.

The solution I implemented is to limit SPACK_DEPENDENCIES to those of a package iff transitive_rpath is False. But we can add another flag, if you prefer.

**Comment by davydden** [comment]
@tgamblin do you have an alternative solution for this header conflict issues or do you agree with using `pkg.transitive_rpaths` to limit `-I/foo/include` flags as well?

**Comment by mpbelhorn** [comment]
@davydden So I just hit this issue today. It appeared to me that Spack is producing correctly ordered inclusions in the compiler invocation:

```
/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I..  -DALL_TERM_DOC -I../src -I../term -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.
8.5/libcerf-1.3-zq6mrk7v3332dz2b6hski5wlstznm3au/include   -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/readline-6.3-7sjjq7bsggwl2mbmoff2ugsw2ddg5ct7/include -I/autofs/nccs-svm1_
sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/harfbuzz-0.9.37-ol2iwboz4kngkxhbt5fkkphfcpfrwow2/include/harfbuzz -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7
-ppc64le/gcc-4.8.5/pango-1.40.3-7giohnpx2t5gs66uyf3mxhqaampoxznv/include/pango-1.0 -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/pixman-0.34.0-kmpenqgbffxqmcvwoetulqlhwdnimwje/inc
lude/pixman-1 -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/pcre-8.39-7mmsejnph3wrxtctrnau6mkap6lmognh/include -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/2
0170224/linux-rhel7-ppc64le/gcc-4.8.5/glib-2.49.4-bbxl7dn7ar2i56rsaxxji4utj5af5jn7/include/glib-2.0 -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/glib-2.49.4-bbxl7dn7ar2i56rsaxxji
4utj5af5jn7/lib/glib-2.0/include -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/libxml2-2.9.4-wftexwcrgapr4j4ymerlcvoxhg23wrtj/include/libxml2 -I/autofs/nccs-svm1_sw/.testing/belho
rn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/zlib-1.2.10-5iwcknmfpwkcwlb745itf6n7y24awsxr/include -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/libpng-1.6.
27-gct4eazredlwf3zobicb4ziyro64jzkw/include/libpng16 -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/freetype-2.7-bjonghd2wvzksxhlwt4s2joj7so6xlp2/include/freetype2 -I/autofs/nccs-s
vm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/fontconfig-2.11.1-r2pzoeuuxnnxfahpgcr7yxw7rks5h3zw/include -I/autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-pp
c64le/gcc-4.8.5/cairo-1.14.8-kkhe35ldysbkeubmtrf2nyzqtb3ihwpk/include/cairo    -g -O2 -c -o alldoc2gih-termdoc.o `test -f 'termdoc.c' || echo './'`termdoc.c
In file included from /usr/include/termios.h:40:0,
                 from /autofs/nccs-svm1_sw/.testing/belhorn/summitdev/.spack/opt/spack/20170224/linux-rhel7-ppc64le/gcc-4.8.5/ncurses-6.0-5kxu47u4irgu2wupfucieuitboerz5hm/include/term.h:86,
                 from doc2x.h:72,
                 from doc2gih.c:57:
/usr/include/bits/termios.h:22:1: error: expected expression before ‘typedef’
 typedef unsigned char cc_t;
 ^
make[3]: *** [doc2gih.o] Error 1
```

In particular, `gcc -DHAVE_CONFIG_H -I. -I..  -DALL_TERM_DOC -I../src -I../term` seems to be the right order placing gnuplot internal headers at higher priority. An alternative to modifying spack would be to patch gnuplot to be less ambiguous about which `term.h` to include:

**gnuplot/package.py** (your version, which is better than what's in the builtin repo):
```
#...
# transitive_rpaths = False # Don't need this if doc2x.h is more explicit.
patch('term_include.patch')
#...
```

**gnuplot/term_include.patch**:
```
--- a/docs/doc2x.h      2017-03-10 13:55:51.719850190 -0500
+++ b/docs/doc2x.h      2017-03-10 13:56:17.569826925 -0500
@@ -69,7 +69,7 @@
 # ifdef ALL_TERM_DOC
 #  include "allterm.h"
 # else
-#  include "term.h"
+#  include "src/term.h"
 # endif
     NULL
 };
```

**Comment by davydden** [comment]
Nice, I will update the PR. Thanks.

**Comment by davydden** [comment]
@adamjstewart @tgamblin this gnuplot fix should be uncontroversial. Thanks to @mpbelhorn there is a simpler solution which does not touch build environment. PR can be squashed to keep the message from the second commit only.

File: var/spack/repos/builtin/packages/gnuplot/package.py [hunk 2]
```diff
@@ -43,24 +41,97 @@ class Gnuplot(Package):
...
+
+    # optional dependencies:
+    depends_on('libcerf', when='+libcerf')
+    depends_on('libgd', when='+gd')
+    depends_on('cairo', when='+cairo')
     depends_on('wx', when='+wx')
+    depends_on('pango', when='+wx')
+    depends_on('pango', when='+cairo')
+    # FIXME: macports add this dependency
+    # depends_on('libxpm', when='+x11')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: I'm not sure if you have access to Linux, but if you build it on Linux, you can run `ldd -r` on the shared libraries/binaries to see whether or not it links to `libxpm`. I'm sure there is an equivalent for macOS.

File: var/spack/repos/builtin/packages/gnuplot/package.py [hunk 3]
```diff
@@ -43,24 +41,97 @@ class Gnuplot(Package):
...
+
+    def configure_args(self):
+        # see https://github.com/Homebrew/homebrew-core/blob/master/Formula/gnuplot.rb
+        # and https://github.com/macports/macports-ports/blob/master/math/gnuplot/Portfile
+        spec = self.spec
+        options = [
+            '--prefix=%s' % prefix,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: This is automatically added by `AutotoolsPackage` so you don't need this line.

File: var/spack/repos/builtin/packages/gnuplot/package.py [hunk 4]
```diff
@@ -43,24 +41,97 @@ class Gnuplot(Package):
...
+            '--disable-dependency-tracking',
+            '--disable-silent-rules',
+            # Per upstream: "--with-tutorial is horribly out of date."
+            '--without-tutorial'
+        ]
+
+        if 'readline' in spec:
+            options.append('--with-readline=%s' % spec['readline'].prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: I assume `--without-readline` isn't accepted?

File: var/spack/repos/builtin/packages/gnuplot/package.py [hunk 5]
```diff
@@ -43,24 +41,97 @@ class Gnuplot(Package):
...
+        else:
+            options.append('--disable-wxwidgets')
+
+        if 'gd' in spec:
+            options.append('--with-gd=%s' % spec['libgd'].prefix)
+        else:
+            options.append('--without-gd')
+
+        if '~cairo' in spec:
+            options.append('--without-cairo')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: Does `--with-cairo` not work?

File: var/spack/repos/builtin/packages/gnuplot/package.py (hunk 2 above)
Comment (davydden) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: look like it does not link to this lib:
```
$ otool -L /Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/gnuplot-5.0.5-omujzkrp7hcqd6ta6weeov47exh6xpoh/bin/gnuplot
/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/gnuplot-5.0.5-omujzkrp7hcqd6ta6weeov47exh6xpoh/bin/gnuplot:
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/readline-6.3-iltuux7p2oywu6wx7zvm5tnsdifge5wh/lib/libreadline.6.dylib (compatibility version 6.0.0, current version 6.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/ncurses-6.0-yz5pwlpnmngucsbfj2aswcowcf2gnav6/lib/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/zlib-1.2.11-ahv5tlvsgqpenkehyj7s6oagr5hbksnv/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libcerf-1.3-ewfiew3edmlveuycspchxoqrmgcqpnu3/lib/libcerf.1.dylib (compatibility version 2.0.0, current version 2.3.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.4.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.0.0)
```

File: var/spack/repos/builtin/packages/gnuplot/package.py (hunk 3 above)
Comment (davydden) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: well aware of that, but the bug still slipped through from the original code :smile:

**Comment by davydden** [comment]
ok, it looks now everything is picked up properly:
```
$ otool -L /Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/gnuplot-5.0.5-omujzkrp7hcqd6ta6weeov47exh6xpoh/bin/gnuplot
/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/gnuplot-5.0.5-omujzkrp7hcqd6ta6weeov47exh6xpoh/bin/gnuplot:
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libcerf-1.3-ewfiew3edmlveuycspchxoqrmgcqpnu3/lib/libcerf.1.dylib (compatibility version 2.0.0, current version 2.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/readline-6.3-iltuux7p2oywu6wx7zvm5tnsdifge5wh/lib/libreadline.6.dylib (compatibility version 6.0.0, current version 6.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/ncurses-6.0-yz5pwlpnmngucsbfj2aswcowcf2gnav6/lib/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/zlib-1.2.11-ahv5tlvsgqpenkehyj7s6oagr5hbksnv/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libgd-2.2.4-me7ixfjderv2zdr5r3nbfef47elpzi7l/lib/libgd.3.dylib (compatibility version 4.0.0, current version 4.4.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libpng-1.6.27-4kcj3ehltx5onnu6jfpki7k7bbvl2ivk/lib/libpng16.16.dylib (compatibility version 44.0.0, current version 44.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/freetype-2.7-s4qsl5knwbnvradqwazmlsajphdjh5wg/lib/libfreetype.6.dylib (compatibility version 19.0.0, current version 19.6.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fontconfig-2.11.1-er2ylppeqpapqsq3ritxybftp6rcxgd2/lib/libfontconfig.1.dylib (compatibility version 10.0.0, current version 10.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/jpeg-9b-exrhsccgrpytkrvf7x4qls4akdpuqcqz/lib/libjpeg.9.dylib (compatibility version 12.0.0, current version 12.0.0)
	/opt/X11/lib/libXpm.4.dylib (compatibility version 16.0.0, current version 16.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libtiff-4.0.6-6ujevx254ifjmhjonqk2wwap36r4vewy/lib/libtiff.5.dylib (compatibility version 8.0.0, current version 8.4.0)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/pango-1.40.3-ritt4gr2ye3gmai3xdpg4or2ypnt7rwf/lib/libpangocairo-1.0.0.dylib (compatibility version 4001.0.0, current version 4001.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/pango-1.40.3-ritt4gr2ye3gmai3xdpg4or2ypnt7rwf/lib/libpango-1.0.0.dylib (compatibility version 4001.0.0, current version 4001.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/glib-2.49.7-pthjviwiefkrqthxbd3sz5ryepgaolc3/lib/libgobject-2.0.0.dylib (compatibility version 4908.0.0, current version 4908.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/cairo-1.14.8-e6lkdxxfmbfikvhrc3sfhl3ujg2z6cvf/lib/libcairo.2.dylib (compatibility version 11403.0.0, current version 11403.8.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/glib-2.49.7-pthjviwiefkrqthxbd3sz5ryepgaolc3/lib/libglib-2.0.0.dylib (compatibility version 4908.0.0, current version 4908.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/gettext-0.19.8.1-mjmtbg2p4w7bzrtlkqcd4ls4r7wbbj4b/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.5.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.4.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.0.0)
```

**Comment by adamjstewart** [comment]
Wait a minute, now it is linking to the system `libxpm`? Can you get it to link to Spack's `libxpm`? You could also try adding the `libiconv` dependency, but I've had back luck with that.

**Comment by davydden** [comment]
> Wait a minute, now it is linking to the system libxpm

right, i did not notice it! Good catch, will do!

**Comment by davydden** [comment]
with the latest version I have
```
$ otool -L /Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/gnuplot-5.0.5-65siw3hguf3xpxf7g6e3kqe6vqqq24pt/bin/gnuplot
/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/gnuplot-5.0.5-65siw3hguf3xpxf7g6e3kqe6vqqq24pt/bin/gnuplot:
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libcerf-1.3-ewfiew3edmlveuycspchxoqrmgcqpnu3/lib/libcerf.1.dylib (compatibility version 2.0.0, current version 2.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/readline-6.3-iltuux7p2oywu6wx7zvm5tnsdifge5wh/lib/libreadline.6.dylib (compatibility version 6.0.0, current version 6.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/ncurses-6.0-yz5pwlpnmngucsbfj2aswcowcf2gnav6/lib/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/zlib-1.2.11-ahv5tlvsgqpenkehyj7s6oagr5hbksnv/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libgd-2.2.4-me7ixfjderv2zdr5r3nbfef47elpzi7l/lib/libgd.3.dylib (compatibility version 4.0.0, current version 4.4.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libpng-1.6.27-4kcj3ehltx5onnu6jfpki7k7bbvl2ivk/lib/libpng16.16.dylib (compatibility version 44.0.0, current version 44.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/freetype-2.7-s4qsl5knwbnvradqwazmlsajphdjh5wg/lib/libfreetype.6.dylib (compatibility version 19.0.0, current version 19.6.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fontconfig-2.11.1-er2ylppeqpapqsq3ritxybftp6rcxgd2/lib/libfontconfig.1.dylib (compatibility version 10.0.0, current version 10.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/jpeg-9b-exrhsccgrpytkrvf7x4qls4akdpuqcqz/lib/libjpeg.9.dylib (compatibility version 12.0.0, current version 12.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libxpm-3.5.11-wbij3y3ngdwo2ptrmyagrnocrkcxw5j5/lib/libXpm.4.dylib (compatibility version 16.0.0, current version 16.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libtiff-4.0.6-6ujevx254ifjmhjonqk2wwap36r4vewy/lib/libtiff.5.dylib (compatibility version 8.0.0, current version 8.4.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/libiconv-1.14-f6wqdlbvf7tsi4nxnhp2p5rloccrb2nw/lib/libiconv.2.dylib (compatibility version 8.0.0, current version 8.1.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/pango-1.40.3-ritt4gr2ye3gmai3xdpg4or2ypnt7rwf/lib/libpangocairo-1.0.0.dylib (compatibility version 4001.0.0, current version 4001.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/pango-1.40.3-ritt4gr2ye3gmai3xdpg4or2ypnt7rwf/lib/libpango-1.0.0.dylib (compatibility version 4001.0.0, current version 4001.3.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/glib-2.49.7-pthjviwiefkrqthxbd3sz5ryepgaolc3/lib/libgobject-2.0.0.dylib (compatibility version 4908.0.0, current version 4908.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/cairo-1.14.8-e6lkdxxfmbfikvhrc3sfhl3ujg2z6cvf/lib/libcairo.2.dylib (compatibility version 11403.0.0, current version 11403.8.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/glib-2.49.7-pthjviwiefkrqthxbd3sz5ryepgaolc3/lib/libglib-2.0.0.dylib (compatibility version 4908.0.0, current version 4908.0.0)
	/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/gettext-0.19.8.1-mjmtbg2p4w7bzrtlkqcd4ls4r7wbbj4b/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.5.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.4.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.0.0)
```

File: var/spack/repos/builtin/packages/gnuplot/package.py [hunk 6]
```diff
@@ -43,24 +41,99 @@ class Gnuplot(Package):
...
+            description='Enable PBM (Portable Bit Map) and other older bitmap terminals')  # NOQA: ignore=E501
+
+    # required dependencies
     depends_on('readline')
-    depends_on('libcerf')
-    depends_on('libgd')
-    depends_on('cairo')
-    depends_on('pango')
+    depends_on('pkg-config', type='build')
+    depends_on('libxpm')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: This may only be required when `+X`. Does it link to `libxpm` even when `--without-x`?

Comment (davydden) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: Yes, it does. I did not build with X

Comment (davydden) [inline var/spack/repos/builtin/packages/gnuplot/package.py]: interestingly enough, macports has it as conditional dependency: https://github.com/macports/macports-ports/blob/master/math/gnuplot/Portfile#L83-L86 But that's not how it works for me on Sierra.

File: var/spack/repos/builtin/packages/wx/package.py [hunk 7]
```diff
@@ -41,18 +42,24 @@ class Wx(Package):
     version('3.0.2', '6461eab4428c0a8b9e41781b8787510484dea800')
     version('3.0.1', '73e58521d6871c9f4d1e7974c6e3a81629fddcf8')
 
+    version('develop', git='https://github.com/wxWidgets/wxWidgets.git', branch='master')
+
     depends_on('gtkplus')
 
-    def make_wx(self):
-        make()
+    parallel = False
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/wx/package.py]: Previously this built 3.1.0+ in parallel. Were you having problems with that?

Comment (davydden) [inline var/spack/repos/builtin/packages/wx/package.py]: No, I just don't think it's possible to keep the old logic with AutotoolsPackage

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/wx/package.py]: You can, just add:
```python 
@when('@:3.0.2')
def build(self, spec, prefix):
    make(parallel=False)
```

Comment (davydden) [inline var/spack/repos/builtin/packages/wx/package.py]: thanks, it works indeed.

**Pull request merged**